packages/webapp/pages/squads/[handle]/[token].tsx

```diff
@@ -74,17 +77,25 @@ const SquadReferral = ({ token, handle }: SquadReferralProps): ReactElement => {
     },
   );
   const sourceId = member?.source?.id;
-  const squadsUrl = `/squads/${sourceId}`;
   const { mutateAsync: onJoinSquad } = useMutation(
     () => joinSquadInvitation({ sourceId, token }),
-    { onSuccess: () => router.replace(squadsUrl) },
+    {
+      onSuccess: (data) => {
+        const squad = squads.find(({ id }) => id === data.id);
+        if (squad) return router.replace(squad.permalink);
```

Comment:
Is this really needed? Not sure...

Reply:
So you won't have to add locally. It would be duplicated in the list if we add it and if in your boot it exists already.